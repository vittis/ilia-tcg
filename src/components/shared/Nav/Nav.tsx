import { cn } from "@/lib/utils";
import { useTranslation } from "react-i18next";
import { Link } from "react-router-dom";

export function Nav() {
	const { t } = useTranslation();

	return (
		<div className={cn("flex flex-col items-center justify-center")}>
			<Link to="/" className="text-glow text-3xl font-bold italic text-yellow-400 md:text-5xl">
				Pokemon TCG
			</Link>
			<div className="text-glow mt-2 text-sm font-bold text-zinc-100">{t("nav.subtitle")}</div>
		</div>
	);
}
