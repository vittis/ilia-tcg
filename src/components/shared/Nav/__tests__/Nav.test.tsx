import { render, screen } from "@testing-library/react";
import { Nav } from "../Nav";
import { BrowserRouter } from "react-router-dom";

describe("Nav", () => {
	it("matches snapshot", () => {
		const { container } = render(
			<BrowserRouter>
				<Nav />
			</BrowserRouter>,
		);

		expect(container).toMatchSnapshot();
	});

	it("should render subtitle", () => {
		render(
			<BrowserRouter>
				<Nav />
			</BrowserRouter>,
		);

		expect(screen.getByText("⚡ Search your favorite cards ⚡")).toBeDefined();
	});
});
