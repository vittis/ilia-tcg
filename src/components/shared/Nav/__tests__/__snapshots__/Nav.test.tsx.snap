// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Nav > matches snapshot 1`] = `
<div>
  <div
    class="flex flex-col items-center justify-center"
  >
    <a
      class="text-glow text-3xl font-bold italic text-yellow-400 md:text-5xl"
      href="/"
    >
      Pokemon TCG
    </a>
    <div
      class="text-glow mt-2 text-sm font-bold text-zinc-100"
    >
      nav.subtitle
    </div>
  </div>
</div>
`;
